boot/zephyr/include/sysflash/sysflash.h

```diff
@@ -3,10 +3,21 @@
 #ifndef __SYSFLASH_H__
 #define __SYSFLASH_H__
 
+#ifdef USE_PARTITION_MANAGER
+#include <pm_config.h>
+
+#define FLASH_AREA_IMAGE_PRIMARY    PM_MCUBOOT_PARTITIONS_PRIMARY_ID
+#define FLASH_AREA_IMAGE_SECONDARY  PM_MCUBOOT_PARTITIONS_SECONDARY_ID
+#define FLASH_AREA_IMAGE_SCRATCH    PM_MCUBOOT_SCRATCH_ID
```

Comment:
Should this be MCUBoot-specific?

Reply:
Do you mean the partitions? Or the code in this file?

No, ideally it should be shared between zephyr and mcuboot. I just had a bug because of this split :) Not sure if changing it is in the scope of this PR-set though.